core/src/main/java/com/amplifyframework/auth/result/AuthSocialSignInResult.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.result;
+
+import androidx.core.util.ObjectsCompat;
+
+public class AuthSocialSignInResult {
```

Comment:
When would someone extend this? Can you suggest a use case in a class-level JavaDoc? Or, alternately, add `final` to close the class?

How come we `AuthSocialSignInResult` needs to be distinct from just an `AuthSignInResult`?

Reply:
While the core value of isSignedIn is common to all sign in results, the nextStep and possible additional data for the results of the different sign in method can be completely different - AWSMobileClient already has a different result object for the federatedSignIn and showSignIn methods than normal sign in and there's reason to believe other plugins would as well.

Reply:
Possibly implement a common interface for isSignedIn